Scripts/Mac/Common-Functions.ps1

```diff
@@ -0,0 +1,153 @@
+<#
+.SYNOPSIS
+Generates the text for GVFS.Installer.Mac nuspec file from the provided inputs
+#>
+function Write-Nuspec
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [string]
+        $PackageVersion,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GvfsInstallerPkg,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GitInstallerPkg,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GcmInstallerPkg
+    )
+
+     $template =
+     "<?xml version=""1.0""?>
+     <package xmlns=""http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd"">
+       <metadata>
+         <id>GVFS.Installers.Mac</id>
+         <version>$PackageVersion</version>
+         <authors>Microsoft</authors>
+         <requireLicenseAcceptance>false</requireLicenseAcceptance>
+         <description>GVFS and G4M Mac installers</description>
+       </metadata>
+       <files>
+         <file src=""$GvfsInstallerPkg"" target=""GVFS"" />
+         <file src=""$GitInstallerPkg"" target=""G4M"" />
+         <file src=""$GcmInstallerPkg"" target=""GCM"" />
+       </files>
+     </package>"
+
+     return $template
+}
+
+<#
+.DESCRIPTION
+Downloads the specified version of GCM Core to the specified DownloadLocation
+#>
+function Download-GCM
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [string]
+        $GcmVersion,
+
+        [Parameter(Mandatory)]
+        [string]
+        $DownloadLocation
+    )
+
+    $packageName = "gcmcore-osx-$GcmVersion.pkg"
+    $url = "https://github.com/microsoft/Git-Credential-Manager-Core/releases/download/v2.0.33-beta/$packageName"
+    $outfile = $DownloadLocation + "/" + $packageName
+
+    Invoke-WebRequest -Uri $url -Outfile $outfile
+
+    return $outfile
+}
+
+<#
+.SYNOPSIS
+Parses the version of the NuGet package that contains the Git installer from a props file.
+#>
+function Get-GitPackageVersionFromProps
+{
+	param
+	(
+		[Parameter(Mandatory)]
+		[string]
+		$GvfsPropsPath
+	)
+
+	$gitPackageVersionLine = Get-ChildItem -Path $GvfsPropsPath | Select-String -Pattern 'GitPackageVersion'
+	$matchFount = $gitPackageVersionLine -match '<GitPackageVersion>(.*)</GitPackageVersion>'
+	return $matches[1]
+}
+
+<#
+.SYNOPSIS
+Given a path to where NuGet packages are located, find the GitForMac NuGetPackage version and extract the Git version.
+#>
+function Get-GitVersionFromNuGetPackage
+{
+	param
+	(
+		[Parameter(Mandatory)]
+		[string]
+		$NuGetPackagesDir,
+
+		[Parameter(Mandatory)]
+		[string]
+		$GitPackageVersion
+	)
+
+	# Find the git version number (looking through packages directory)
+	$gitInstallerPath = $NuGetPackagesDir + "/gitformac.gvfs.installer/" + $GitPackageVersion + "/tools"
```

Comment:
This whole function is not used - but I think it could still be valuable if we wanted to add an extra layer of validation (at some point). This script depends on the correct version of git being places in the `$PackagesDir` directory, but this script does not validate that expectation.

We could use this function to ensure the expected git version is being consumed by finding the git installer version that the Git Installer NuGet package contains, and make sure it matches the git installer version in the `$PackagedDir`